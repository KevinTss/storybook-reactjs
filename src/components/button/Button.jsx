import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const StyledButton = styled.button`
  background-color: white;
  border: 1px solid black;
  color: black;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-flex;
  font-size: 16px;
  cursor: pointer;

  ${({ type }) => {
    if (type === 'primary') {
      return `
        background-color: #4caf50;
        border-color: #4caf50;
        color: white;
      `
    }
  }}
`

class Button extends Component {
  constructor(props) {
    super(props)
  }

  handleClick = e => {
    e.preventDefault()
    console.log('Clicked', this)
  }

  render() {
    return (
      <StyledButton type={this.props.type} onClick={this.handleClick}>
        {this.props.label}
      </StyledButton>
    )
  }
}

Button.propTypes = {
  type: PropTypes.oneOf(['default', 'primary']),
}

Button.defaultProps = {
  type: 'default',
}

export default Button
