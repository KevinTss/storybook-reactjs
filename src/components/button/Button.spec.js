import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
// import App, { Counter, dataReducer } from './App';
import Button from './Button'

describe('hoy', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Button label='Hello' />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
