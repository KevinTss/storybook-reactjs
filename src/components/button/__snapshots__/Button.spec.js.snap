// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hoy snapshot renders 1`] = `
<button
  className="Button__StyledButton-f9aaxx-0 cEMXbr"
  onClick={[Function]}
  type="default"
>
  Hello
</button>
`;
